src/actions/motion.ts

```diff
@@ -59,6 +59,37 @@ abstract class MoveByScreenLine extends BaseMovement {
       let start = Position.FromVSCodePosition(vimState.editor.selection.start);
       let stop = Position.FromVSCodePosition(vimState.editor.selection.end);
       let curPos = Position.FromVSCodePosition(vimState.editor.selection.active);
+      let anchorPos = Position.FromVSCodePosition(vimState.editor.selection.anchor);
+      let prevPos =
+        this.movementType === 'up'
+          ? curPos.getDown(curPos.character)
+          : curPos.getUp(curPos.character);
+      if (prevPos.line === start.line && prevPos.isAfter(start)) {
+        prevPos = prevPos.getLeft();
+      }
+
+      // Make cursor moves properly
```

Comment:
I haven't dug into this yet, but can you explain why these adjustments are necessary to make the cursor move correctly?

Reply:
1. cursor position is before the anchor and the next position is also before the anchor

   - gk/gj works properly

1. cursor position is after the anchor and the next position is also after the anchor

   - gk - moves cursor upper-right
   - gj - moves cursor lower-right

1. cursor position is after the anchor and the next position is before the anchor

   - gk - anchor position moves left itself

1. cursor position is before the anchor and next position is after the anchor

   - gj - anchor position moves right itself

This PR fixes 2-4 behaviors.

By the way, when I reading the code, I found that behavior 2 is caused by the below code.

https://github.com/VSCodeVim/Vim/blob/086011322c281465f6bdd6e95d2e31fb69b5cfdd/src/mode/modeHandler.ts#L532-L540

So I fixed this.